internal/handlers/common/update.go

```diff
@@ -110,6 +120,55 @@ func UpdateDocument(doc, update *types.Document) (bool, error) {
 	return changed, nil
 }
 
+// processCurrentdateFieldExpression changes document according to $currentDate operator.
+// If the document was changed it returns true.
+func processCurrentdateFieldExpression(doc *types.Document, currentDateVal any) (bool, error) {
+	var changed bool
+	var err error
+	currentDateExpression := currentDateVal.(*types.Document)
+
+	now := time.Now()
+	sort.Strings(currentDateExpression.Keys())
```

Comment:
Why do the keys need to be sorted?

Reply:
> Why do the keys need to be sorted?

The addition of the new keys influences the order inside the document. 
The process of document fields has to be in lexicographic order.